torch/_dynamo/compile_package.py

```diff
@@ -0,0 +1,756 @@
+
+    def add_backward_aoti(self, aoti: "CompiledAOTI") -> None:
+        assert self.backward_aoti is None
+        self.backward_aoti = aoti
+
+    def add_aot_autograd(self, aot_autograd: "AOTAutogradCacheEntry") -> None:
+        assert self.aot_autograd is None
+        self.aot_autograd = aot_autograd
+
+    def add_aot_config(self, aot_config: "AOTConfig") -> None:
+        assert self.aot_config is None
+        self.aot_config = aot_config
+
+    def update_aot_autograd_joint(self, aot_autograd: "AOTAutogradCacheEntry") -> None:
+        assert self.aot_autograd is not None
+        assert aot_autograd.compiled_bw is not None
+        self.aot_autograd = aot_autograd
+
+    def add_compile_fx_kwargs(self, compile_fx_kwargs) -> None:
+        assert self.compile_fx_kwargs is None
+        self.compile_fx_kwargs = compile_fx_kwargs
+
+    @functools.cached_property
+    def _callable(self):
+        class PrecompiledFunction:
+            def __init__(self, callback):
+                self.callback = callback
+
+            def load(self, *args, **kwargs):
+                return self.callback
+
+            def post_compile(self, compiled_func, *args, **kwargs):
+                # TODO Does this work if we just remove the boxing?
+                if not hasattr(compiled_func, "_boxed_call"):
+                    from torch._functorch._aot_autograd.utils import make_boxed_func
+
+                    compiled_func = make_boxed_func(compiled_func)
+                return compiled_func
+
+        entry = dataclasses.replace(
+            self.aot_autograd, compiled_fw=PrecompiledFunction(self.forward_aoti)
+        )
+        if self.backward_aoti is not None:  # TODO Test this.
+            entry = dataclasses.replace(
+                entry, compiled_bw=PrecompiledFunction(self.backward_aoti)
+            )
+
+        # TODO logging should be removed? but currently needed to make this not throwing
+        with torch._dynamo.utils.dynamo_timed(
+            "backend_compile", log_pt2_compile_event=True
+        ):
+            compiled_fn = entry.wrap_post_compile(
+                None, self.aot_config, self.compile_fx_kwargs
+            )
+
+        @torch._dynamo.decorators.disable
+        def forward(*args):
+            return compiled_fn(list(args))
+
+        return forward
+
+
+def _load_graph_compile(load_path: str, graph_compile: _GraphCompile) -> None:
+    """
+    precondition: graph_compile is empty.
+    """
+    from torch._inductor.output_code import CompiledAOTI
+
+    root = os.path.join(load_path, graph_compile.name)
+
+    with open(root + ".aot_autograd", "rb") as f:
+        aot_autograd = pickle.load(f)
+    graph_compile.add_aot_autograd(aot_autograd)
+
+    with open(root + ".aot_config", "rb") as f:
+        from torch._functorch._aot_autograd.autograd_cache import AOTConfig
+
+        aot_config = pickle.load(f)
+
+        def _(*args, **kwargs):
+            raise RuntimeError("NYI")
```

Comment:
You're never gonna be able to serialize these functions, so they're not NYI, right? They're actually full forward/backward compilers(often inductor libraries). Instead I think you want to ignore these callables (or assume that they are always the same)